client/packages/requisitions/src/ResponseRequisition/DetailView/IndicatorsTab.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import {
+  Box,
+  ButtonWithIcon,
+  DataTableSkeleton,
+  IndicatorLineRowNode,
+  NothingHere,
+  PlusCircleIcon,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { ProgramIndicatorFragment, ResponseFragment } from '../api';
+
+interface IndicatorTabProps {
+  onClick: (
+    programIndicator?: ProgramIndicatorFragment,
+    indicatorLine?: IndicatorLineRowNode,
+    response?: ResponseFragment
+  ) => void;
+  isLoading: boolean;
+  response?: ResponseFragment;
+  indicators?: ProgramIndicatorFragment[];
+}
+
+export const IndicatorsTab = ({
+  onClick,
+  isLoading,
+  response,
+  indicators,
+}: IndicatorTabProps) => {
+  const t = useTranslation();
+  if (isLoading) {
+    return <DataTableSkeleton />;
+  }
+  if (!indicators || indicators.length === 0) {
+    return <NothingHere body={t('error.no-indicators')} />;
+  }
+  const regimenIndicators = indicators.filter(
+    indicator => indicator.code === 'REGIMEN'
+  );
+  const hivIndicators = indicators.filter(
+    indicator => indicator.code === 'HIV'
+  );
+
+  return (
+    <Box display="flex" flexDirection="column" padding={2} gap={2}>
+      {regimenIndicators.length > 0 && (
+        <ButtonWithIcon
+          // disabled={disableAddButton}
+          label={t('button.regimen')}
+          Icon={<PlusCircleIcon />}
+          onClick={() =>
+            onClick(
+              regimenIndicators[0],
+              regimenIndicators[0]?.lineAndColumns[0]?.line,
+              response
+            )
+          }
+        />
+      )}
+      {hivIndicators.length > 0 && (
```

Comment:
![Screenshot 2024-11-27 at 1 16 32 PM](https://github.com/user-attachments/assets/31bec5ba-3e34-4589-b9b8-976fad3a5233)

Still seeing HIV indicators on my non-HIV program?

Reply:
Hmm I think if you look at an old program requisition, the button would still be there but if you click it everything gets filtered out? It's getting a bit gross cause I already do the same filter in `IndicatorEditPage` and `IndicatorLineEdit`, and I'd have to do it again in `IndicatorTab`. In hindsight I really should have persisted with doing it in the backend, as this will pop up as a problem request requisitions too.

As discussed I've also been ignoring that you can turn HIV off on program and the indicators will still be there 😭. I'm not 100% sure if OG does it, but ideally OG would set the HIV program_indicator to be inactive:

https://github.com/msupply-foundation/open-msupply/blob/97f0eeecfcc247d733fe8099eaaa103863b391b8/server/repository/src/db_diesel/program_indicator_row.rs#L22

I don't believe we're doing any filtering on that yet but.